compiler-core/src/javascript.rs

```diff
@@ -810,3 +813,56 @@ fn bool(bool: bool) -> Document<'static> {
         false => "false".to_doc(),
     }
 }
+
+/// Int segments <= 48 bits wide in bit arrays are within JavaScript's safe range and are evaluated
+/// at compile time when all inputs are known. This is done for both bit array expressions and
+/// pattern matching.
+///
+/// Int segments of any size could be evaluated at compile time, but currently aren't due to the
+/// potential for causing large generated JS for inputs such as `<<0:8192>>`.
+///
+pub(crate) const SAFE_INT_SEGMENT_MAX_SIZE: usize = 48;
+
+/// Evaluates the value of an Int segment in a bit array into its corresponding bytes. This avoids
+/// needing to do the evaluation at runtime when all inputs are known at compile-time.
+///
+pub(crate) fn bit_array_segment_int_value_to_bytes(
+    mut value: BigInt,
+    size: BigInt,
+    endianness: endianness::Endianness,
+    location: SrcSpan,
+) -> Result<Vec<u8>, Error> {
+    // Clamp negative sizes to zero
+    let size = size.max(BigInt::ZERO);
+
+    // The segment size in bits is not allowed to exceed the range of a u32. At runtime the
+    // limit is lower than this and depends on the JS engine. V8's limit is currently
+    // 2^30-1 bits.
+    let size = size.to_u32().ok_or_else(|| Error::Unsupported {
+        feature: "Integer segment size greater than 2^32-1".into(),
```

Comment:
It would be breaking if this function was called with integers of any size, but it's limited to those up to `SAFE_INT_SEGMENT_MAX_SIZE` as ints of arbitrary size would result in enormous generated code from input such as `<<0:1_000_000>>`.

The comment and error handling here was misleading and has been fixed.

Reply:
Thank you